src/snowflake/connector/result_set.py

```diff
@@ -130,32 +127,37 @@ def _fetch_arrow_batches(
                 f"Trying to use arrow fetching on {head_type} which "
                 f"is not ArrowResultChunk"
             )
-        return self._create_iter(iter_unit=TABLE_UNIT)
 
-    def _fetch_arrow_all(self):
+    def _fetch_arrow_batches(
+        self,
+    ) -> Iterator[Table]:
+        """Fetches all the results as Arrow Tables, chunked by Snowflake back-end."""
+        self.__can_create_arrow_iter()
+        return self._create_iter(iter_unit=IterUnit.TABLE_UNIT, structure="arrow")
+
+    def _fetch_arrow_all(self) -> Optional[Table]:
```

Comment:
We should also expose a `_fetch_arrow_batches` similar to how `_fetch_pandas_batches` works.

Also idk if this is strictly part of this ticket, but I want to expose `fetch_arrow_batches` and `fetch_arrow_all` through the regular `cursor`. Could you please add this too?

Reply:
You already have this code in the PR. I must have forgot the first half of the code review, sorry! 😅